src/api.js

```diff
@@ -513,81 +527,79 @@ const history: HandlerFunction = async function (req, _res) {
   }
   const verifiedBody = utils.validateHistoryReq(addressesRequestLimit, apiResponseLimit, input);
 
-  switch (verifiedBody.kind) {
-    case "ok": {
-      const body = verifiedBody.value;
-      const limit = apiResponseLimit;
-      const [referenceTx, referenceBlock] = (body.after && [body.after.tx, body.after.block]) || [undefined, undefined];
-      const referenceBestBlock = body.untilBlock;
-
-      const afterBlockNum = await askBlockNum(referenceBlock, referenceTx != undefined ? referenceTx : "");
-      const untilBlockNum = await askBlockNum(referenceBestBlock);
-
-      if (afterBlockNum.kind === 'error') {
-        if (afterBlockNum.errMsg.includes('tx')) {
-          throw new Error("REFERENCE_TX_NOT_FOUND");
-        }
-        if (afterBlockNum.errMsg.includes('block')) {
-          throw new Error("REFERENCE_BLOCK_MISMATCH");
-        }
-        return { status: 400, body: afterBlockNum.errMsg}
-      }
-      if (untilBlockNum.kind === 'error') {
-        throw new Error("REFERENCE_BEST_BLOCK_MISMATCH");
-      }
+  if (verifiedBody.kind === 'error') {
+    return { status: 400, body: verifiedBody.errMsg };
+  }
+  if (verifiedBody.kind !== 'ok') {
+    return utils.assertNever(verifiedBody);
+  }
```

Comment:
There was a `switch` with three branches: if the message is `ok`, if the message is `error` and default. I have just changed if to have two IFs for the `error` and for the NOT `ok` branches, and then the longest `ok` branch turns into the default function branch after the IFs and it's much easier to read